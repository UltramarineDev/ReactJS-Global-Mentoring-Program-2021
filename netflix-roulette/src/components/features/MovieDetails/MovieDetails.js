import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { wordings } from '/src/locales/wordings';
import MovieImage from '/src/components/shared/MovieImage/MovieImageContainer';
import Header from '/src/components/shared/Header/Header';
import StoryCard from '/src/components/shared/StoryCard/StoryCard';
import MoviesList from '/src/components/shared/MoviesList/MoviesListContainer';

import styles from './MovieDetails.module.scss';

const MovieDetails = ({ movie, year, onClick }) => {
  const duration = `${movie.runtime} ${wordings.min}`;
  
  return (
    <>
      <StoryCard gradientClassName={styles.gradientClassName}>
        <div className={styles.root}>
          <Header>
            <a onClick={onClick} className={styles.icon}>
              <FontAwesomeIcon icon="search"/>
            </a>
          </Header>
          <div className={styles.movieDetails}>
            <MovieImage imageUrl={movie.poster_path} />
            <div className={styles.description}>
              <div className={styles.title}>{movie.title}</div>
              <div className={styles.tagLine}>{movie.tagline}</div>
              <div className={styles.dataWrapper}>
                <span className={styles.year}>{year}</span>
                <span>{duration}</span>
              </div>
              <div>{movie.overview}</div>
            </div>
          </div>
       </div>
      </StoryCard>
      <MoviesList />
    </>
)};

MovieDetails.propTypes = {
  movie: PropTypes.object.isRequired,
  year: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default MovieDetails;