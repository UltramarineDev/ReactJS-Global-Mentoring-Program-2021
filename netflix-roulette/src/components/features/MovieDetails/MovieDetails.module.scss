.root {
  position: relative;
  padding: 24px 64px;
  z-index: 1;
}

.icon {
  color: #f65261;
  transform: rotate(90deg);
  font-size: 25px;
  background-color: unset;
  border: none;
}

.gradientClassName {
  background-image: linear-gradient(0deg, rgba(0, 0, 0, 0.8) 60%, rgba(0, 0, 0, 0.9) 60%);
}

.movieDetails {
  display: flex;
  flex-direction: row;
  justify-content: start;
  align-items: flex-start;
  margin-top: 30px;
  max-width: 90%;
}

.titleWrapper {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  padding-bottom: 12px;
  padding-top: 24px;
}

.title {
  padding-right: 16px;
  text-transform: uppercase;
  font-size: 46px;
}

.description {
  padding-left: 42px;
}

.tagLine {
  padding-bottom: 24px;
}

.dataWrapper {
  padding-bottom: 24px;
  color: #f65261;
  font-size: 26px;
}

.year {
  padding-right: 32px;
}

.rating {
  border-radius: 50%;
  width: 42px;
  min-width: 42px;
  height: 42px;
  padding: 8px;
  border: 2px solid #666;
  text-align: center;
  font-size: 26px;
  color: #80B859;
}
