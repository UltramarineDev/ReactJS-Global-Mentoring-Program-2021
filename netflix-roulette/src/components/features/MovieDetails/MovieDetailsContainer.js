import React, { PureComponent } from 'react';

import { data } from '../../shared/MoviesList/data.json';
import MovieDetails from './MovieDetails';

class MovieDetailsContainer extends PureComponent {
   state = {
    movie: data[3],
  }

  handleClick = () => {};

  render() {
    const { movie } = this.state;

    const year = movie ? new Date(movie.release_date).getFullYear() : '-';
    return (
      <MovieDetails
        movie={movie}
        year={year}
        onClick={this.handleClick}
      />
    );
  }
};

export default MovieDetailsContainer;
