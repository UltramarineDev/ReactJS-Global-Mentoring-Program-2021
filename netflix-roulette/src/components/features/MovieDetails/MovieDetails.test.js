import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { useSelector } from 'react-redux';
import { render, cleanup, screen } from '@testing-library/react';
import MovieDetails from './MovieDetails';
import '@testing-library/jest-dom/extend-expect';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

jest.mock('../../shared/MoviesList/MoviesList', () => 'MovieList');

describe('MovieDetails', () => {
  afterEach(() => {
    cleanup();
  });
  test('renders MovieDetails', () => {
    const state = { fetchMovies: { movie: { } } };
    useSelector.mockImplementation(() => state);

    const component = renderer.create(
      <MemoryRouter>
        <MovieDetails />
      </MemoryRouter>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders MovieDetails', () => {
    renderer.create(
      <MemoryRouter>
        <MovieDetails />
      </MemoryRouter>,
    );

    expect(mockDispatch).toHaveBeenCalledTimes(1);
  });

  test('renders MovieDetails', () => {
    const state = { fetchMovies: { movie: { runtime: 100, vote_average: 9 } } };
    useSelector.mockImplementation(() => state);
    
    render(
      <MemoryRouter>
        <MovieDetails />
      </MemoryRouter>,
    );

    expect(screen.getByText('9')).toBeInTheDocument();
  });
});
