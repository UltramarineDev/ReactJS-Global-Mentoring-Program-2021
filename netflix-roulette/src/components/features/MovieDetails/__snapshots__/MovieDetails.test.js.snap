// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders MovieDetails 1`] = `
Array [
  <div
    className="storyCard"
  >
    <div
      className="background"
    >
      <img
        alt="Background image"
        src="https://assets.nflxext.com/ffe/siteui/vlv3/e178a4e7-4f52-4661-b2ae-41efa25dca7c/eb618394-24cd-43ae-8799-c1a76a7bcbfc/BY-en-20210222-popsignuptwoweeks-perspective_alpha_website_small.jpg"
      />
      <div
        className="gradient gradientClassName"
      />
    </div>
    <div
      className="root"
    >
      <div
        className="header"
      >
        <span
          className="title"
        >
          <strong>
            netflix
          </strong>
          roulette
        </span>
        <a
          className="icon"
          href="/"
          onClick={[Function]}
        />
      </div>
      <div
        className="movieDetails"
      >
        <div
          className="noImageWrapper"
        />
        <div
          className="description"
        >
          <div
            className="titleWrapper"
          >
            <span
              className="title"
            />
          </div>
          <div
            className="tagLine"
          />
          <div
            className="dataWrapper"
          >
            <span
              className="year"
            >
              NaN
            </span>
            <span>
              
            </span>
          </div>
          <div />
        </div>
      </div>
    </div>
  </div>,
  <MovieList />,
]
`;
